### DISCUSSION TITLE
Station setup page (UI improvement)

### DISCUSSION THREAD

**Discussion by R1BLH** [wavelog/wavelog#502] [Ideas]
The Station Settings page now shows all logs and station locations. This is not a bad thing.
But there is one problem I just ran into. When you change the active log, you have to change the active station location. As a result, it is possible to select a station location that is not linked to the active log.

It would be nice if the station locations that are bound to the active log were somehow highlighted, this would reduce the likelihood of the above situation. 
An additional plus would be if the selection of an unlinked station and log would be highlighted in a different color.

![изображение](https://github.com/wavelog/wavelog/assets/16667173/d78f7aa8-d95b-492a-8fcc-336d6eb3c58d)

**Comment by HB9HIL** [comment]
What do you think @AndreasK79? The stationsetup is your baby :)

**Comment by AndreasK79** [comment]
@R1BLH I am all for improvements here, and that sounds like a good improvement. My plan with station setup was to make it easier to do the setup, and have it in one page. We did see that many users had trouble with understanding how to do this, or how it worked.

I am not sure how to solve your feature request yet, but I will give it some thought.

**Reply by R1BLH** [reply]
@AndreasK79  in a normal desktop application, everything is simply implemented.
On the OnLoad or OnRefresh event, the program checks which log is active and the background of the table rows is colored according to whether these rows belong to the active log, then the display is refreshed.
Then, on any OnClick event (on the log table or the station location table), the check is repeated, the background of the rows is changed, and the display is refreshed.
I don't know if this is possible on the web.

**Reply by AndreasK79** [reply]
@R1BLH the tables are loaded dynamically, and reloaded on change. However, no checks are done between them, and if logbook is changed, the location do not reload. So I guess that the loading of locations needs to check if they are in the active logbook, and mark that (color or flag) in the table. Perhaps it is not so difficult to add something like that.

**Reply by R1BLH** [reply]
@AndreasK79this is exactly the same as a thinked about!

**Reply by AndreasK79** [reply]
I need to fix the backend code, but this is what I have in mind for the locations:
![image](https://github.com/wavelog/wavelog/assets/6977712/ca14b489-65c7-43ce-90ca-40237c6298a9)
![image](https://github.com/wavelog/wavelog/assets/6977712/ed4996ae-fe8a-4f99-9478-10211a1221ff)

**Reply by R1BLH** [reply]
It looks good!

**Reply by AndreasK79** [reply]
If you want to test, here is the PR: https://github.com/wavelog/wavelog/pull/533

**Reply by R1BLH** [reply]
Ok, thanks!
